### PR TITLE
feat: v6.5 compatibility

### PR DESCRIPTION
**Pull request by rittou**
This MR modifies the code for compatibility with Shopware version 6.5

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     }
   },
   "require": {
-    "shopware/core": "~6.4.0",
-    "shopware/storefront":  "~6.4.0"
+    "shopware/core": "6.5.x-dev",
+    "shopware/storefront":  "6.5.x-dev"
```
Comment (elkmod) [inline composer.json:16]: Shouldn't this refer to a stable version and not the main/dev branch?

Comment (rittou) [reply composer.json:16]: The stable version is NOT available yet, only the RC version is available now.
Maybe we will wait until the stable is available.

Comment (elkmod) [reply composer.json:16]: Okay, I would say for merging we should wait, yes.

Comment (mkucmus) [reply composer.json]: hey guys! I think we are ready to set the dependency for `^6.5.0.0` now 🚀 
I will proceed with the topic in order to make the PWA working well with the latest release as well -> https://github.com/vuestorefront/shopware-pwa/issues/1992

Comment (rittou) [reply composer.json]: @mkucmus I updated it with the official 6.5.0.0.

File: composer.json [hunk 2]
```diff
@@ -12,8 +12,8 @@
     }
   },
   "require": {
-    "shopware/core": "~6.4.0",
-    "shopware/storefront":  "~6.4.0"
+    "shopware/core": "6.5.0.0",
+    "shopware/storefront":  "6.5.0.0"
```
Comment (BrocksiNet) [inline composer.json:16]: Why not?
```suggestion
    "shopware/core": "~6.5.0",
    "shopware/storefront":  "~6.5.0"
```

Comment (rittou) [reply composer.json:16]: My mistake.

**Review by mkucmus — COMMENTED** [review]
docs: compatibility table

**Review by mkucmus — APPROVED** [review]
🌎 has been waiting for that!

**Pull request merged**